pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -722,7 +714,12 @@ func blackholeFilterChain(push *model.PushContext, node *model.Proxy) *listener.
 		},
 		Filters: append(
 			buildMetricsNetworkFilters(push, node, istionetworking.ListenerClassSidecarOutbound),
-			blackholeFilters...,
+			&listener.Filter{
```

Comment:
This is replacing a global (blackholeFilters), which was causing unexpected repeated patching when used with envoyFilter, with an inline definition.

What about other globals, where are they defined?

Reply:
There are lot in `pilot/pkg/xds/filters/filters.go`. Probably others was well. I think these are either much less common to mutate or possible since they are nested another layer compared to TcpProxy they are not impacted. Not 100% sure.

This was definitely not intended to be super-robust fix everything PR. Just a first step